src/components/ClusterManager.js

```diff
@@ -262,6 +262,15 @@ export default ajaxCaller(class ClusterManager extends PureComponent {
     }
   }
 
+  async executeAndRefreshWithNav(promise) {
+    const { namespace, name } = this.props
+
+    this.executeAndRefresh(promise)
+    if (/notebooks\/.+/.test(window.location.hash)) {
```

Comment:
What does this line do?

Reply:
It makes the navigation happen only if you're on a notebook or terminal, not the notebooks list or anywhere else